adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,225 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.rdd.features
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.{ SequenceRecord, SequenceDictionary, Coverage, ReferenceRegion }
+import org.bdgenomics.adam.rdd.GenomicRDD
+
+import scala.annotation.tailrec
+
+object CoverageRDD {
+
+  /**
+   * @param elem The coverage element to extract a sequence record from.
+   * @return Gets the SequenceRecord for coverage.
+   */
+  private def getSequenceRecord(elem: Coverage): SequenceRecord = {
+    SequenceRecord(elem.contigName, 1L)
+  }
+
+  /**
+   * Builds a CoverageRDD without SequenceDictionary information by running an
+   * aggregate to rebuild the SequenceDictionary.
+   *
+   * @param rdd The underlying Coverage RDD to build from.
+   * @return Returns a new CoverageRDD.
+   */
+  private[rdd] def apply(rdd: RDD[Coverage]): CoverageRDD = {
+
+    // aggregate to create the sequence dictionary
+    val sd = new SequenceDictionary(rdd.map(getSequenceRecord)
+      .distinct
+      .collect
+      .toVector)
+
+    CoverageRDD(rdd, sd)
+  }
+}
+
+case class CoverageRDD(rdd: RDD[Coverage],
+                       sequences: SequenceDictionary) extends GenomicRDD[Coverage, CoverageRDD] {
+
+  /**
+   * Saves coverage as feature file.
+   * @see FeatureRDD.save
+   *
+   * Supported file formats include bed, narrowPeak and parquet. Coverage is saved
+   * as a feature where coverage is stored in score attribute.
+   *   val chrom    = feature.getContigName
+   *   val start    = feature.getStart
+   *   val end      = feature.getEnd
+   *   val name     = Features.nameOf(feature)
+   *   val coverage = feature.getScore
+   *
+   * @param filePath The location to write the output.
+   */
+  def save(filePath: java.lang.String) = {
+    this.toFeatureRDD.save(filePath)
+  }
+
+  /**
+   * Merges adjacent ReferenceRegions with the same value. This is used for merging coverage.
```

Comment:
> This is used for merging coverage.

This isn't really detailed. Split it out of the summary line and expand upon the explanation.